tests/e2e/workflows/workflow_test.go

```diff
@@ -0,0 +1,294 @@
+	resp, err := utils.HTTPPost(postString, nil)
+	if err != nil {
+		return fmt.Sprintf("Failure starting workflow: %s", err.Error())
+	}
+
+	getString := fmt.Sprintf("%s/dapr/%s", url, instanceID)
+	resp, err = utils.HTTPGet(getString)
+	if err != nil {
+		return fmt.Sprintf("Failure getting info on newly started workflow: %s", err.Error())
+	}
+	if string(resp) != "Running" {
+		return fmt.Sprintf("Expected workflow to be Running, actual workflow state is: %s", string(resp))
+	}
+
+	// Terminate the workflow
+	postString = fmt.Sprintf("%s/TerminateWorkflow/dapr/%s", url, instanceID)
+	resp, err = utils.HTTPPost(postString, nil)
+	if err != nil {
+		return fmt.Sprintf("Failure terminating workflow: %s", err.Error())
+	}
+	resp, err = utils.HTTPGet(getString)
+	if err != nil {
+		return fmt.Sprintf("Failure getting info on terminated workflow: %s", err.Error())
+	}
+	if string(resp) != "Terminated" {
+		return fmt.Sprintf("Expected workflow to be Terminated, actual workflow state is: %s", string(resp))
+	}
+
+	// Purge the workflow
+	postString = fmt.Sprintf("%s/PurgeWorkflow/dapr/%s", url, instanceID)
+	resp, err = utils.HTTPPost(postString, nil)
+	if err != nil {
+		return fmt.Sprintf("Failure purging workflow: %s", err.Error())
+	}
+
+	// Startup a new workflow with the same instanceID to ensure that it is available
+	postString = fmt.Sprintf("%s/StartWorkflow/dapr/placeOrder/%s", url, instanceID)
+	resp, err = utils.HTTPPost(postString, nil)
+	if err != nil {
+		return fmt.Sprintf("Failure starting workflow: %s", err.Error())
+	}
+
+	instanceID = string(resp)
+	resp, err = utils.HTTPGet(getString)
+	if err != nil {
+		return fmt.Sprintf("Failure getting info on newly started workflow: %s", err.Error())
+	}
+	if string(resp) != "Running" {
+		return fmt.Sprintf("Expected workflow to be Running, actual workflow state is: %s", string(resp))
+	}
+
+	return string(resp)
+}
+
+var workflowAppTests = []struct {
+	in               string
+	instanceID       string
+	app              string
+	expectedResponse string
+}{
+	{
+		"start",
+		"startID",
+		"workflowsapp",
+		"Running",
+	},
+
+	{
+		"pauseResume",
+		"pauseID",
+		"workflowsapp",
+		"Success",
+	},
+
+	{
+		"purge",
+		"purgeID",
+		"workflowsapp",
+		"Running",
+	},
+
+	{
+		"raiseEvent",
+		"raiseEventID",
+		"workflowsapp",
+		"Completed",
+	},
+}
+
+func TestWorkflow(t *testing.T) {
+	for _, tt := range workflowAppTests {
+		t.Run(tt.in, func(t *testing.T) {
+			// Get the ingress external url of test app
+			externalURL := tr.Platform.AcquireAppExternalURL(tt.app)
+			require.NotEmpty(t, externalURL, "external URL must not be empty")
+
+			// Check if test app endpoint is available
+			_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+			require.NoError(t, err)
+
+			// Sleep so that the workflow engine starts
+			time.Sleep(5 * time.Second)
+
+			result := "false"
+			switch tt.in {
+			case "purge":
+				result = purgeTest(externalURL, tt.instanceID)
```

Comment:
Although this works, and I am not going to ask you to re-write this test, it's a rather unusual way to write a test, a bit anti-pattern..

Normally you'd create a function that accepts `t *testing.T` as first parameter, and then you'd use assertions such as `assert.Equal` or `require.Equal` inside to make the test fail. For future tests, please consider following this pattern.